crates/client/starknet-block-import/src/compilation.rs

```diff
@@ -0,0 +1,179 @@
+use blockifier::execution::contract_class::{ContractClass as BlockifierCasmClass, EntryPointV1};
+use cairo_lang_starknet::contract_class::{
+    ContractClass as SierraContractClass, ContractEntryPoint as SierraEntryPoint,
+    ContractEntryPoints as SierraEntryPoints,
+};
+use cairo_lang_starknet::contract_class_into_casm_contract_class::StarknetSierraCompilationError;
+use cairo_lang_utils::bigint::BigUintAsHex;
+use cairo_vm::types::program::Program;
+use cairo_vm::types::relocatable::MaybeRelocatable;
+use mc_rpc::casm_contract_class_to_compiled_class;
+use mp_felt::Felt252Wrapper;
+use num_bigint::BigUint;
+use starknet_api::deprecated_contract_class::EntryPointType as DeprecatedEntryPointType;
+use starknet_api::hash::StarkFelt;
+use starknet_api::state::{
+    ContractClass as BlockifierSierraClass, EntryPoint as BlockifierEntryPoint,
+    EntryPointType as BlockifierEntryPointType,
+};
+use starknet_core::types::contract::{
+    CompiledClass, CompiledClassEntrypoint, CompiledClassEntrypointList, ComputeClassHashError,
+};
+use starknet_core::types::{FieldElement, FromByteArrayError};
+
+#[derive(Debug, thiserror::Error)]
+pub enum CompilationError {
+    #[error(transparent)]
+    ComputeClassHash(#[from] ComputeClassHashError),
+    #[error(transparent)]
+    SierraCompilation(#[from] StarknetSierraCompilationError),
+    #[error("Contract class V0 is not supported")]
+    UnsupportedClassV0,
+    #[error("Unexpected relocatable while converting program to bytecode")]
+    UnexpectedRelocatable,
+    #[error("Failed to parse felt from bytes: {0}")]
+    FeltFromBytes(#[from] FromByteArrayError),
+}
+
+pub(crate) fn blockifier_casm_class_to_compiled_class_hash(
+    casm_class: BlockifierCasmClass,
+) -> Result<FieldElement, CompilationError> {
+    match casm_class {
+        BlockifierCasmClass::V0(_) => Err(CompilationError::UnsupportedClassV0),
```

Comment:
we can easily support that no?
It would be useful for full nodes working with the whole chain history like deoxys.
@antiyro wdyt?